### DISCUSSION TITLE
How to use bootstrap and persist only read-only credential to pull from the repo?

### DISCUSSION THREAD

**Discussion by SebTardif** [fluxcd/flux2#3121] [Q&A]
Hi,

I used flux for 2 years, and my brain still hurt each time I use bootstrap because it seems I cannot use a read-only credential to pull from the repo.

I don't know how to but it seems I should be able to:
- Use bootstrap to generate yaml files but not execute, and not write to the git repo, so that I have the yaml I need to put in my Infrastructure as Code framework (IaC) and so also doesn't write in my repo.
- Pass two set of credentials to bootstrap, one to write its yaml to the repo, and also one to be persisted in k8s for the read-only pulling.

Any advice?

**Comment by stefanprodan** [comment]
Bootstrap by default generates a read only deploy key. It uses the GitHub, GitLab, BitBucket PAT to push the manifests from your local machine, while Flux in-cluster uses SSH with the read-only access. Docs here: https://fluxcd.io/flux/installation/#github-and-github-enterprise

**Reply by airtonix** [reply]
I did all that but everytime i tear the cluster down and re-setup it up, a bootstrap results in flux commiting changes to my setup that i do not want.

command i use to bootstrap: https://github.com/airtonix/clusterfiles/blob/1965a7ffb544f51a2a99e5cab9888ab7b476f12d/cli/commands/bootstrap.sh#L40-L56
